core/services/src/main/java/org/exoplatform/services/rest/AttachmentsRestService.java

```diff
@@ -0,0 +1,306 @@
+/**
+ *
+ */
+package org.exoplatform.services.rest;
+
+import io.swagger.annotations.*;
+import org.apache.commons.lang.StringUtils;
+import org.exoplatform.common.http.HTTPStatus;
+import org.exoplatform.commons.exception.ObjectNotFoundException;
+import org.exoplatform.services.attachments.model.Attachment;
+import org.exoplatform.services.attachments.rest.model.AttachmentEntity;
+import org.exoplatform.services.attachments.service.AttachmentsService;
+import org.exoplatform.services.attachments.utils.EntityBuilder;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.rest.resource.ResourceContainer;
+import org.exoplatform.social.core.manager.IdentityManager;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Path("/attachments")
+@Api(value = "/attachments", description = "Managing attachments")
+public class AttachmentsRestService implements ResourceContainer {
+  private static final Log LOG = ExoLogger.getLogger(AttachmentsRestService.class.getName());
+
+  protected AttachmentsService attachmentsService;
+
+  protected IdentityManager identityManager;
+
+  public AttachmentsRestService(AttachmentsService attachmentsService,
+                                IdentityManager identityManager) {
+    this.attachmentsService = attachmentsService;
+    this.identityManager = identityManager;
+  }
+
+  /**
+   * Link an existing attachments to the given entity (Event, Task, Wiki,...)
+   *
+   * @param entityId the entity id.
+   * @param entityType the entity type.
+   * @param attachmentIds the list of attachments uuid to be linked.
+   * @return The response.
+   * @throws Exception The exception
+   * @anchor ManageDocumentService.linkAttachmentsToEntity
```

Comment:
No need to add javadoc API in REST endpoints since you have added swager comments